magda-web-client/src/Components/Dataset/DatasetPage.tsx

```diff
@@ -31,6 +33,11 @@ const DatasetPage: FunctionComponent<PropsType> = props => {
 
     const publisherId = dataset?.publisher?.id ? dataset.publisher.id : null;
 
+    const editButtonTooltipText =
+        "The `Edit` function is only available to datasets created by Magdad.";
+
+    const isDatasetEditable = dataset.identifier?.indexOf("magda-") === 0;
```

Comment:
I'm not comfortable with this as the detection mechanism. If we ever change the way that we generate ids for internal datasets (e.g. what if we renamed the product?) it'll break this :(. 

Aren't we storing the system that created the metadata in an aspect somewhere? If so we should start for Magda.

Reply:
😄 was also wondering whether we should do it in a more formal way.
How about we create the `source` aspect for all datasets that are created by Magda.
And check whether a dataset is an internal dataset by looking at this aspect?
In this way, we can safely assume that if a dataset has no `source` aspect, it must NOT be an internal one